qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,438 @@
+            cutoff_date: Retrieve the most recent parameter up until the cutoff date.
+                Parameters generated after the cutoff date will be ignored. If the
+                cutoff_date is None then all parameters are considered.
+
+        Returns:
+            value: The value of the parameter.
+
+        Raises:
+            CalibrationError: if there is no parameter value for the given parameter name
+                and pulse channel.
+        """
+        param_name = param.name if isinstance(param, Parameter) else param
+        sched_name = schedule.name if isinstance(schedule, Schedule) else schedule
+
+        if (sched_name, param_name) not in self._parameter_map:
+            raise CalibrationError(f"No parameter for {param_name} and schedule {sched_name}.")
+
+        param = self._parameter_map[(sched_name, param_name)]
+
+        if qubits not in self._params[param]:
+            raise CalibrationError(f"No parameter value for {param} and qubits {qubits}.")
+
+        if valid_only:
+            candidates = [p for p in self._params[param][qubits] if p.valid]
+        else:
+            candidates = self._params[param][qubits]
+
+        candidates = [candidate for candidate in candidates if candidate.group == group]
+
+        if cutoff_date:
+            candidates = [val for val in candidates if val.date_time <= cutoff_date]
+
+        if len(candidates) == 0:
+            msg = (
+                f"No candidate parameter values for {param_name} in calibration group "
+                f"{group} on qubits {qubits} in schedule {sched_name} "
+            )
+
+            if cutoff_date:
+                msg += f" Cutoff date: {cutoff_date}"
+
+            raise CalibrationError(msg)
+
+        candidates.sort(key=lambda x: x.date_time)
+
+        return candidates[-1].value
+
+    def get_schedule(
+        self,
+        name: str,
+        qubits: Tuple[int, ...],
+        free_params: List[str] = None,
+        group: Optional[str] = "default",
+    ) -> Schedule:
+        """
+        Args:
+            name: The name of the schedule to get.
+            qubits: The qubits for which to get the schedule.
+            free_params: The parameters that should remain unassigned.
+            group: The calibration group from which to draw the
+                parameters. If not specifies this defaults to the 'default' group.
+
+        Returns:
+            schedule: A deep copy of the template schedule with all parameters assigned
+                except for those specified by free_params.
+
+        Raises:
+            CalibrationError: if the name of the schedule is not known.
+        """
+
+        # Get the schedule and deepcopy it to prevent binding from removing
+        # the parametric nature of the schedule.
+        if name not in self._schedules:
+            raise CalibrationError("Schedule %s is not defined." % name)
+
+        sched = copy.deepcopy(self._schedules[name])
```

Comment:
Deepcopy is no longer necessary. Now you can set `inplace=False` to assign method.

Reply:
Nice, see here: 0862722